src/routes/blog/post/build-fullstack-notes-app-cursor-appwrite-tanstack-start/+page.markdoc

```diff
@@ -0,0 +1,396 @@
+export { ID };
+```
+
+---
+
+# Create Database & Collection via MCP
+
+**Prompt (Cursor):**
+```text
+Using the Appwrite MCP server, create a database "main" and a collection "notes" with fields:
+title, content, userId, createdAt, updatedAt.
+Grant authenticated user read/write and save IDs to src/lib/config.ts.
+```
+
+---
+
+# Add Authentication (session-safe, no auto-login)
+
+**Prompt (Cursor):**
+```text
+Create an auth helper with session-safe login/logout:
+
+Requirements:
+- login(email, password):
+  1. Check if a session exists; if yes, delete all sessions first.
+  2. Create a new email/password session.
+  3. If "session already active" error occurs, delete all sessions and retry once.
+  4. Return the current user via account.get().
+
+- logout(): delete all sessions.
+
+- getCurrentUser(): return account.get(); if unauthenticated, return null.
+
+Update the /login page:
+- NEVER auto-redirect away from /login. Always show the form so users can switch accounts.
+- On successful login, navigate to /notes.
+- Show clear errors if something fails.
+
+Update route guards:
+- /notes loader/server check: if no session, redirect to /login.
+- / index loader/server check: if session → /notes, else → /login.
+```
+
+**Helper (`src/lib/auth.ts`):**
+```ts
+import { Account, Client, ID } from 'appwrite';
+
+const client = new Client()
+  .setEndpoint(import.meta.env.VITE_APPWRITE_ENDPOINT!)
+  .setProject(import.meta.env.VITE_APPWRITE_PROJECT_ID!);
+
+const account = new Account(client);
+
+export async function getCurrentUser() {
+  try {
+    return await account.get();
+  } catch {
+    return null;
+  }
+}
+
+export async function deleteAllSessions() {
+  try {
+    await account.deleteSessions();
+  } catch {
+    // ignore; we might not have a session
+  }
+}
+
+export async function login(email: string, password: string) {
+  await deleteAllSessions();
+
+  try {
+    await account.createEmailPasswordSession(email, password);
+  } catch (err: any) {
+    const msg = err?.message?.toLowerCase?.() ?? '';
+    if (msg.includes('session') && msg.includes('active')) {
+      await deleteAllSessions();
+      await account.createEmailPasswordSession(email, password);
+    } else {
+      throw err;
+    }
+  }
+
+  return await account.get();
+}
+
+export async function signup(email: string, password: string, name?: string) {
+  await account.create(ID.unique(), email, password, name);
+  return await login(email, password);
+}
+
+export async function logout() {
+  await deleteAllSessions();
+}
+```
+
+---
+
+# Add CRUD for Notes
+
+**Prompt (Cursor):**
+```text
+Create a /notes route that lists, creates, updates, and deletes notes using Appwrite Databases API.
+Include loading, error, and empty states.
+```
+
+---
+
+# Style and Quality Pass (Tailwind v4, Accessibility)
+
+**Prompt (Cursor):**
+```text
+Add Tailwind CSS v4 and basic styling. 
+Fix any invalid utilities like focus-visible:ring-opacity-* or custom @apply utilities.
```

Comment:
Should we mention why this line is needed in the prompt? Maybe that it's an error you encountered?